### PR TITLE
shrink mypy exclude list

### PR DESCRIPTION
**Pull request by danieleades**

### CODE REVIEW

File: tests/test_builders/test_build_html.py [hunk 1]
```diff
@@ -1,8 +1,10 @@
 """Test the HTML builder and check output against XPath."""
+from __future__ import annotations
```
Comment (AA-Turner) [inline tests/test_builders/test_build_html.py:3]:
```suggestion

from __future__ import annotations
```

File: tests/test_directives/test_directive_object_description.py [hunk 2]
```diff
@@ -1,4 +1,7 @@
 """Test object description directives."""
+from __future__ import annotations
```
Comment (AA-Turner) [inline tests/test_directives/test_directive_object_description.py:3]:
```suggestion

from __future__ import annotations
```

**Review by AA-Turner — APPROVED** [review]
In general, is it worth type-checking the test files? It seems like it will be a a lot of effort to make work (we have a massive blacklist of files, as seen here), with questionable potential gain?

I might be missing something though...

Regardless, this looks good as always.

A

**Comment by danieleades** [comment]
> In general, is it worth type-checking the test files? It seems like it will be a a lot of effort to make work (we have a massive blacklist of files, as seen here), with questionable potential gain?

i think it's hugely valuable. Apart from catching errors in those files themselves, it also helps to capture non-local changes. For an example, a change in an interface in the module that the test file is testing might get caught by type checking of the test file (or conversely might be hidden by 'excluding' the file)

case in point- there are a bunch of re-exports in the 'util' module. It's not clear from looking at this folder if they are all used or not. If the test directory had 100% mypy coverage, you could delete these and mypy would immediately tell you about this missing items. As it is, you don't find out about missing items until you run the tests and they fail

**Pull request merged**